### PR TITLE
Add ability to inject a field based on matching pattern

### PR DESCRIPTION
**Pull request by mlasevich**
This allows to add a field to record based on which pattern was matched.

(May likely address #19)

### CODE REVIEW

**Comment by mlasevich** [comment]
@repeatedly Any change we can get this merged? Thanks.

File: lib/fluent/plugin/parser_multi_format.rb [hunk 1]
```diff
@@ -1,34 +1,50 @@
 require 'fluent/plugin/parser'
 
+SubParser = Struct.new(:parser, :index, :name)
+FORMAT_NAME = "format_name"
```
Comment (repeatedly) [inline lib/fluent/plugin/parser_multi_format.rb:4]: These definitions are not needed. Following approach is enough.

```
# in def initialize
@parser_format_names = []

# in def configure

@parser_format_names << e.delete('format_name') ...

# in def parse
@parsers.each_with_index { |parser, i|
  ...
  unless @format_key
    record[@format_key] = @parser_format_names[i]
  end
```

**Review by repeatedly — COMMENTED** [review]
I think `index_key` feature is useless unlike `format_key`.

```
{"key":"value", "parser_index":"0"}  # Hard to handle this value on processing side
```

Implementing only `format_key` is enough for typical use cases.

File: lib/fluent/plugin/parser_multi_format.rb [hunk 2]
```diff
@@ -1,34 +1,50 @@
...
       Plugin.register_parser('multi_format', self)
+      config_param :format_key, :string, default: ''
+      config_param :index_key, :string, default: ''
 
       def initialize
         super
 
         @parsers = []
+        @format_key = ''
+        @index_key = ''
```
[the remark this answers is not in the document]
Comment (mlasevich) [reply lib/fluent/plugin/parser_multi_format.rb]: updated

**Comment by mlasevich** [comment]
@repeatedly Updated as per comments

File: README.md [hunk 3]
```diff
@@ -86,6 +86,46 @@ Use `format` instead of `<parse></parse>`.
...
+      key_name message
+      <parse>
+        @type multi_format
+
+        # if set, add this key to record with value being pattern format name
+        # (format_name key)
+        format_key 'format'
+
+        # If set, add index of the matching pattern as this key (0-based)
+        index_key 'parser_index'
```
Comment (repeatedly) [inline README.md:109]: Could you remove this parts?

Comment (mlasevich) [reply README.md:109]: forgot about the readme

**Comment by repeatedly** [comment]
Sorry for the late. Merged and released v1.1.0 :)

**Pull request merged**